webapp/src/Service/ImportExportService.php

```diff
@@ -83,15 +99,47 @@ public function getContestYamlData(Contest $contest, bool $includeProblems = tru
         return $data;
     }
 
+    protected function convertImportedTime(array $fields, array $data, ?string &$message = null): ?DateTimeImmutable
```

Comment:
Would be good to briefly comment what this does: find the first set field from `$fields` in `$data` and parse it as a date.
Also, would be good to clarify that there's a distinction betwee a return value of `null` and parse failure, and that that's only the case when `$message` is not `null`. Would be better to rename that to `$error_message` or so.